druid/src/lib.rs

```diff
@@ -26,7 +26,8 @@ mod command;
 mod data;
 mod env;
 mod event;
-mod lens;
+pub mod lens;
```

Comment:
I think we don't want the modules in the namespace, just the items such as `Lens` and so on. I can see some possible issues with the intermediate structs, in which case we should probably be using `pub(crate)` if `impl Lens` will work as a type, or failing that, `#[doc(hidden)]`.

Reply:
`impl Lens` is unfortunately impossible in an extension trait for the time being. I'm not sure how I feel about going out of our way to hide these types; what's the upside?

Reply:
It's not so much that we want to hide the types, as that we want to avoid modules being exported directly into the namespace. This is recent reorg work by @cmyr and maybe he has stronger feelings about it.

Reply:
Personally, I like `pub mod foo { struct Bar; }` when I would otherwise have to do `struct FooBar;`, which I think is the case here; the lens names are pretty vague out of context.

Reply:
I'm 100% okay with _some_ public modules in the case where there is a clear argument for them, e.g. they have tons of exports and it is clarifying to keep them separate, and this sounds like a reasonable case for that.

I mostly just don't want public modules to be the default choice.